cli/cmd/check.go

```diff
@@ -62,11 +65,15 @@ non-zero exit code.`,
 	cmd.Args = cobra.NoArgs
 	cmd.PersistentFlags().StringVar(&options.versionOverride, "expected-version", options.versionOverride, "Overrides the version used when checking if Linkerd is running the latest version (mostly for testing)")
 	cmd.PersistentFlags().BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
-	cmd.PersistentFlags().BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
+	cmd.PersistentFlags().StringVar(&options.targetProxyResource, "proxy", options.targetProxyResource, "Only run data-plane checks, to determine if the data plane is healthy")
 	cmd.PersistentFlags().DurationVar(&options.wait, "wait", options.wait, "Maximum allowed time for all tests to pass")
 	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
 	cmd.PersistentFlags().BoolVar(&options.singleNamespace, "single-namespace", options.singleNamespace, "When running pre-installation checks (--pre), only check the permissions required to operate the control plane in a single namespace")
 
+	if cmd.Flags().Changed("proxy") {
```

Comment:
This unfortunately will not ever be true because the check is run before `flag.Parse()` is executed. and `cmd.Flags().Changed` documentation describes the function as:
```
// Changed returns true if the flag was explicitly set during Parse() and false
// otherwise
```
It looks like moving this to `validate()` would have the same effect.
```diff
diff --git a/cli/cmd/check.go b/cli/cmd/check.go
index ad1c8a17..305967ef 100644
--- a/cli/cmd/check.go
+++ b/cli/cmd/check.go
@@ -137,6 +137,10 @@ func configureAndRunChecks(w io.Writer, options *checkOptions) error {
 }
 
 func (o *checkOptions) validate() error {
+	if o.targetProxyResource != "" {
+		o.dataPlaneOnly = true
+	}
+
 	if o.preInstallOnly && o.dataPlaneOnly {
 		return errors.New("--pre and --proxy flags are mutually exclusive")
 	}

```

Reply:
🤔 chmm, but your proposed solution does not end up doing what I want it to be doing because I need for --proxy to work even without any target specified. So basically for it to keep working as it is working now when --proxy is just a boolean flag. Without target specified, it will target everything.

Anyway it looks like I should be able to do it in `configureAndRunChecks` which is after `Parse` call so I fixed that.

Reply:
Ah great. That works even better! 👍